text/0000-target-tier-policy.md

```diff
@@ -0,0 +1,685 @@
+  - This requirement exists to ensure that Rust builds, including nightly
+    builds, can meet the necessary requirements to allow users to smoothly run
+    the host tools.
+- Providing host tools does not exempt a target from requirements to support
+  cross-compilation if at all possible.
+- All tier 2 requirements apply.
+
+A target may be promoted directly from tier 3 to tier 2 with host tools if it
+meets all the necessary requirements, but doing so may introduce substantial
+additional complexity. If in doubt, the target should qualify for tier 2
+without host tools first.
+
+## Tier 1 target policy
+
+At this tier, the Rust project guarantees that a target builds and passes all
+tests, and will reject patches that fail to build or pass the testsuite on a
+target. We hold tier 1 targets to our highest standard of requirements.
+
+A proposed new tier 1 target must be reviewed and approved by the compiler team
+based on these requirements. In addition, the release team must approve the
+long-term viability of the target, and the additional work of supporting the
+target. For a tier 1 target, this will typically take place via a full RFC
+proposing the target, to be jointly reviewed and approved by the compiler team
+and release team.
+
+In addition, the infrastructure team must approve the integration of the target
+into Continuous Integration (CI), and the tier 1 CI-related requirements. This
+review and approval may take place in a PR adding the target to CI, by an
+infrastructure team member reporting the outcome of a team discussion, or by
+including the infrastructure team in the RFC proposing the target.
+
+- Tier 1 targets must have substantial, widespread interest within the
+  developer community, and must serve the ongoing needs of multiple production
+  users of Rust across multiple organizations or projects. These requirements
+  are subjective, and determined by consensus of the approving teams. A tier 1
+  target may be demoted or removed if it becomes obsolete or no longer meets
+  this requirement.
+- The target must build and pass tests reliably in CI.
+  - The target must not disable an excessive number of tests or pieces of tests
+    in the testsuite in order to do so. This is a subjective requirement.
+  - If the target does not have host tools support, or if the target has low
+    performance, the infrastructure team may choose to have CI cross-compile
+    the testsuite from another platform, and then run the compiled tests
+    either natively or via accurate emulation. However, the approving teams may
+    take such performance considerations into account when determining the
+    viability of the target or of its host tools.
+- The target must provide as much of the Rust standard library as is feasible
+  and appropriate to provide. For instance, if the target has a standard memory
+  allocator, it must provide an implementation of `alloc` and the associated
+  data structures.
```

Comment:
(same here)

Reply:
I meant "standard" as "target-supplied" here; I didn't realize this might be ambiguous.

In any case, I don't think the target tier policy should be prescriptive about *how* to implement `alloc` for a target. I've reworded this accordingly.